add dimmer-configure-magit

tomkoelman opened the pull request:
I added ` *transient*` to `dimmer-buffer-exclusion-regexps`

Commit:
add dimmer-configure-magit

gonewest818 commented:
Thank you!  Do you mind updating the README and CHANGELOG as well?

tomkoelman committed:
Add dimmer-configure-magit to CHANGELOG and README

tomkoelman commented:
> Thank you! Do you mind updating the README and CHANGELOG as well?

I just added a new commit with those changes. I didn't know what the bracketed number (e.g. [#37]) should be so I just left it out.

gonewest818 commented:
Thank you, totally fine. If there had been a reported issue it would be that number.  But it’s not worth opening a ticket just to close it again.



> On Mar 2, 2020, at 11:52 AM, tomkoelman <notifications@github.com> wrote:
> 
> Thank you! Do you mind updating the README and CHANGELOG as well?
> 
> I just added a new commit with those changes. I didn't know what the bracketed number (e.g. [#37 <https://github.com/gonewest818/dimmer.el/issues/37>]) should be so I just left it out.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub <https://github.com/gonewest818/dimmer.el/pull/38?email_source=notifications&email_token=AASYZ5WN65DXWT6FMBN4MJ3RFQE6HA5CNFSM4K7SAXDKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOENQWPPA#issuecomment-593586108>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AASYZ5Q6HC6RTFPYK3OU2QLRFQE6HANCNFSM4K7SAXDA>.
>

tomkoelman commented:
Got it. I like the package, thanks for sharing.

> On 2 Mar 2020, at 20:59, Neil Okamoto <notifications@github.com> wrote:
> 
> Thank you, totally fine. If there had been a reported issue it would be that number. But it’s not worth opening a ticket just to close it again.
> 
> 
> 
> > On Mar 2, 2020, at 11:52 AM, tomkoelman <notifications@github.com> wrote:
> > 
> > Thank you! Do you mind updating the README and CHANGELOG as well?
> > 
> > I just added a new commit with those changes. I didn't know what the bracketed number (e.g. [#37 <https://github.com/gonewest818/dimmer.el/issues/37>]) should be so I just left it out.
> > 
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub <https://github.com/gonewest818/dimmer.el/pull/38?email_source=notifications&email_token=AASYZ5WN65DXWT6FMBN4MJ3RFQE6HA5CNFSM4K7SAXDKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOENQWPPA#issuecomment-593586108>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AASYZ5Q6HC6RTFPYK3OU2QLRFQE6HANCNFSM4K7SAXDA>.
> > 
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/gonewest818/dimmer.el/pull/38?email_source=notifications&email_token=AAJCZDAW3GDEPJ2WD54HLUTRFQFZ7A5CNFSM4K7SAXDKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOENQXJAY#issuecomment-593589379>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAJCZDF3LA2J6OESCII22YLRFQFZ7ANCNFSM4K7SAXDA>.
>

gonewest818 commented on README.md:
```diff
@@ -52,8 +52,8 @@ ensure org-mode buffers are not dimmed.
 users to ensure posframe buffers are not dimmed.
 
-* `dimmer-configure-which-key` is a convenience function for which-key
-users to ensure which-key popups are not dimmed.
```
It seems you accidentally deleted the documentation for dimmer-configure-which-key. 

And actually, I'm trying to keep these dimmer-configure-xxx functions in alphabetical order, so can this move up to the appropriate place in the list?  Same for the header comment in the source file.

gonewest818 commented on README.md:
```diff
@@ -52,8 +52,8 @@ ensure org-mode buffers are not dimmed.
-* `dimmer-configure-which-key` is a convenience function for which-key
-users to ensure which-key popups are not dimmed.
+* `dimmer-configure-magitis a convenience function for magit users to
+ensure transients are not dimmed.
```
typo: you're missing the closing quote and a space is needed after `dimmer-configure-magit`

gonewest818 commented on dimmer.el:
```diff
@@ -264,6 +267,13 @@ uses a buffer name that fits this pattern."
+;;;###autoload
+(defun dimmer-configure-magit ()
+  "Convenience settings for magit users."
+  (with-no-warnings
```
Technically we don't need a `with-no-warnings` here. There are cases where we need to refer to predicate functions in other packages and the bytecompiler will complain that it doesn't recognize the symbol. But here the regexp doesn't depend on symbols in the package.

gonewest818 reviewed (changes requested):
A few minor changes noted in the comments. Thank you again for contributing!

tomkoelman committed:
fix README

  - dimmer-configure-which-key was erroneously deleted
  - dimmer-configure-magit was mis-typed
  - convenvience functions weren't sorted alphabetically anymore

tomkoelman committed:
remove unneeded `with-no-warnings`

tomkoelman committed:
sort convenience functions in dimmer.el header alphabetically

tomkoelman commented:
Thanks for the detailed feedback. I think I addressed every comment you made in the last three commits.


> On 2 Mar 2020, at 21:12, Neil Okamoto <notifications@github.com> wrote:
> 
> @gonewest818 requested changes on this pull request.
> 
> A few minor changes noted in the comments. Thank you again for contributing!
> 
> In README.md <https://github.com/gonewest818/dimmer.el/pull/38#discussion_r386620348>:
> 
> > -* `dimmer-configure-which-key` is a convenience function for which-key
> -users to ensure which-key popups are not dimmed.
> It seems you accidentally deleted the documentation for dimmer-configure-which-key.
> 
> And actually, I'm trying to keep these dimmer-configure-xxx functions in alphabetical order, so can this move up to the appropriate place in the list? Same for the header comment in the source file.
> 
> In README.md <https://github.com/gonewest818/dimmer.el/pull/38#discussion_r386621068>:
> 
> > +* `dimmer-configure-magitis a convenience function for magit users to
> +ensure transients are not dimmed.
> typo: you're missing the closing quote and a space is needed after dimmer-configure-magit
> 
> In dimmer.el <https://github.com/gonewest818/dimmer.el/pull/38#discussion_r386623518>:
> 
> > @@ -264,6 +267,13 @@ uses a buffer name that fits this pattern."
>      (add-to-list
>       'dimmer-prevent-dimming-predicates #'which-key--popup-showing-p)))
>  
> +;;;###autoload
> +(defun dimmer-configure-magit ()
> +  "Convenience settings for magit users."
> +  (with-no-warnings
> Technically we don't need a with-no-warnings here. There are cases where we need to refer to predicate functions in other packages and the bytecompiler will complain that it doesn't recognize the symbol. But here the regexp doesn't depend on symbols in the package.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/gonewest818/dimmer.el/pull/38?email_source=notifications&email_token=AAJCZDAEDUQ5K6NEZF4RBLTRFQHK7A5CNFSM4K7SAXDKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCXTXCQA#pullrequestreview-367489344>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAJCZDGPLHIUGYLLI3TKYGTRFQHK7ANCNFSM4K7SAXDA>.
>

gonewest818 commented:
Merged!   Glad you like the package!
